### PR TITLE
Added documentation for shared Hugging Face cache..

### PR DESCRIPTION
**Pull request by andriy-safe-ai**

### CODE REVIEW

File: _posts/2023-05-03-cais-cluster-documentation.md [hunk 1]
```diff
@@ -136,7 +137,51 @@ This process does not apply to the /public_models folder, which should only be u
...
+tokenizer = AutoTokenizer.from_pretrained(model_path)
+```
+
+_**Requesting Models and Datasets to be Added to the Cache**_  
+If you require a model or dataset that is not already cached, you can request it by posting in the #shared-models-data Slack channel. Please include the full name, such as `meta-llama/Meta-Llama-3.1-8B`. Requests are typically processed within 24 hours, subject to approval.
+
+You can view the list of models currently available in the global cache on our [Shared Models and Datasets Tracker](https://docs.google.com/spreadsheets/d/1lF4rx-edniZAon5ExcS8-qa-nLdOxb7tTLHXQ0dLubw/edit?pli=1&gid=0#gid=0).
+
+_**Cache Maintenance and Updates**_  
+The global cache is updated regularly, with popular models added automatically based on usage patterns. Older or less frequently used models may be removed periodically to free up space. If you rely on a specific model, ensure it remains in the cache by notifying the admins via Slack.
```
Comment (steven-basart) [inline _posts/2023-05-03-cais-cluster-documentation.md:167]: This should give time frames and be less vague. 
Models unused for over 90 days may be removed at our discretion. Something like that.

File: _posts/2023-05-03-cais-cluster-documentation.md [hunk 2]
```diff
@@ -136,7 +137,51 @@ This process does not apply to the /public_models folder, which should only be u
...
+```python
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+HF_CACHE = "/data/public_models/huggingface/"
+model_path = HF_CACHE + "meta-llama/Meta-Llama-3-8B-Instruct"
+
+print("Loading model...")
+model = AutoModelForCausalLM.from_pretrained(model_path, device_map="auto")
+tokenizer = AutoTokenizer.from_pretrained(model_path)
```
Comment (steven-basart) [inline _posts/2023-05-03-cais-cluster-documentation.md:158]: Should maybe add an example for using a dataset in this same code

File: _posts/2023-05-03-cais-cluster-documentation.md [hunk 3]
```diff
@@ -136,7 +137,51 @@ This process does not apply to the /public_models folder, which should only be u
...
+
+You can view the list of models currently available in the global cache on our [Shared Models and Datasets Tracker](https://docs.google.com/spreadsheets/d/1lF4rx-edniZAon5ExcS8-qa-nLdOxb7tTLHXQ0dLubw/edit?pli=1&gid=0#gid=0).
+
+_**Cache Maintenance and Updates**_  
+The global cache is updated regularly, with popular models added automatically based on usage patterns. Older or less frequently used models may be removed periodically to free up space. If you rely on a specific model, ensure it remains in the cache by notifying the admins via Slack.
+
+_**Troubleshooting**_  
+Here are some common issues and how to resolve them:  
+  - Model Not Found: Double-check the model path for typos. Refer to the Shared Models and Datasets Tracker for the correct path.  
+  - Missing Dependencies: Ensure all required Python packages are installed. Use pip install transformers torch to install the basic dependencies.
```
Comment (steven-basart) [inline _posts/2023-05-03-cais-cluster-documentation.md:172]: Don't suggest `pip install` and just refer to the transformer installation page. Less likely for our docs to be out of date that way (unless the installation page changes).

File: _posts/2023-05-03-cais-cluster-documentation.md [hunk 4]
```diff
@@ -136,7 +137,51 @@ This process does not apply to the /public_models folder, which should only be u
...
+You can view the list of models currently available in the global cache on our [Shared Models and Datasets Tracker](https://docs.google.com/spreadsheets/d/1lF4rx-edniZAon5ExcS8-qa-nLdOxb7tTLHXQ0dLubw/edit?pli=1&gid=0#gid=0).
+
+_**Cache Maintenance and Updates**_  
+The global cache is updated regularly, with popular models added automatically based on usage patterns. Older or less frequently used models may be removed periodically to free up space. If you rely on a specific model, ensure it remains in the cache by notifying the admins via Slack.
+
+_**Troubleshooting**_  
+Here are some common issues and how to resolve them:  
+  - Model Not Found: Double-check the model path for typos. Refer to the Shared Models and Datasets Tracker for the correct path.  
+  - Missing Dependencies: Ensure all required Python packages are installed. Use pip install transformers torch to install the basic dependencies.  
+  - Missing Token: Ensure you have an authentication token for models that require them. Consult the Hugging Face documentation for further information.
```
Comment (steven-basart) [inline _posts/2023-05-03-cais-cluster-documentation.md:173]: Again provide the link to the page.

File: _posts/2023-05-03-cais-cluster-documentation.md [hunk 5]
```diff
@@ -136,7 +137,51 @@ This process does not apply to the /public_models folder, which should only be u
...
+
+For additional help, reach out via the #shared-models-data Slack channel.
```
Comment (steven-basart) [inline _posts/2023-05-03-cais-cluster-documentation.md:175]: Why not help desk?

File: _posts/2023-05-03-cais-cluster-documentation.md [hunk 6]
```diff
@@ -136,7 +137,51 @@ This process does not apply to the /public_models folder, which should only be u
...
+
+_**Using Custom Models**_  
+If you need to cache a custom model locally, feel free to make use of a local HF cache. However, be mindful of storage quotas and only use local caching when necessary.
+
+_**Data Storage and Retention**_  
+
+  - Storage Quotas: Requested models and datasets added to the shared cache do NOT count against your file system quota.  
+  - Retention Policy: Models in the global cache may be removed if they are no longer frequently used. Important models can be flagged for retention by contacting the admins.
```
Comment (steven-basart) [inline _posts/2023-05-03-cais-cluster-documentation.md:183]: Again add time frames.

**Review by steven-basart — CHANGES_REQUESTED** [review]
Some changes requested.

File: _posts/2023-05-03-cais-cluster-documentation.md (hunk 1 above)
Comment (andriy-safe-ai) [reply _posts/2023-05-03-cais-cluster-documentation.md]: Updated.

File: _posts/2023-05-03-cais-cluster-documentation.md (hunk 2 above)
Comment (andriy-safe-ai) [reply _posts/2023-05-03-cais-cluster-documentation.md]: We are no longer adding datasets to the cache.

**Pull request merged**